FoodBookApp/UI/Views/Bookmarks/BookmarksView.swift

```diff
@@ -22,36 +24,63 @@ struct BookmarksView: View {
     }
     
     var body: some View {
-        
-        VStack {
-            // FIXME: the following content is temporary
-            Image(systemName: "book")
-                .foregroundColor(/*@START_MENU_TOKEN@*/.blue/*@END_MENU_TOKEN@*/)
-                .font(.system(size: 100))
-            
-            Button(action: {
-                Task {
-                    do {
-                        print("signing out...")
-                        try AuthService.shared.signOut()
-                        showSignInView = true
-                        
-                        //TODO: move this to sign out view if created
-                        notify.cancelNotification(identifier: "lastReviewNotification")
-                        
-                    } catch {
-                        print("Failed to sign out...")
+        if isAuthenticated {
+            VStack {
+                // FIXME: the following content is temporary
+                Image(systemName: "book")
+                    .foregroundColor(/*@START_MENU_TOKEN@*/.blue/*@END_MENU_TOKEN@*/)
+                    .font(.system(size: 100))
+                
+                Button(action: {
+                    Task {
+                        do {
+                            print("signing out...")
+                            try AuthService.shared.signOut()
+                            showSignInView = true
+                            
+                            //TODO: move this to sign out view if created
+                            notify.cancelNotification(identifier: "lastReviewNotification")
+                            
+                        } catch {
+                            print("Failed to sign out...")
+                        }
                     }
+                }, label: {
+                    Text("Sign out")
+                })
+                .buttonStyle(.borderedProminent)
+                .padding()
+                Text(user?.name ?? "")
+                Text(user?.email ?? "")
+            }
+        } else {
+            VStack {
+                Text("Please confirm it's you to proceed")
+                    .font(.title)
+                    .padding()
+                Button("Authenticate") {
+                    authenticateUser()
                 }
-            }, label: {
-                Text("Sign out")
-            })
-            .buttonStyle(.borderedProminent)
-            .padding()
-            Text(user?.name ?? "")
-            Text(user?.email ?? "")
+                .padding()
+            }
+        }
+    }
+    
+    private func authenticateUser() {
```

Comment:
i think this should be in the View Model or as a helper function somewhere else

Reply:
on it!